### PR TITLE
Validate `morpho` address in constructor

### PR DESCRIPTION
**Pull request by MerlinEgalite**
Fixes https://github.com/cantinasec/review-morpho-blue-1/issues/86

### CODE REVIEW

File: test/forge/DeploymentTest.sol
Comment (Rubilmax) [inline test/forge/DeploymentTest.sol:1]: 👍
much better than mixing deployment tests with other tests

File: test/forge/DeploymentTest.sol [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    function testDeployMetaMorpho(
+        address owner,
+        address morpho,
+        uint256 initialTimelock,
+        string memory name,
+        string memory symbol
+    ) public {
+        assumeNotZeroAddress(owner);
```
Comment (Rubilmax) [inline test/forge/DeploymentTest.sol:27]: is it part of forge-std now?

File: test/forge/DeploymentTest.sol [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    function testDeployMetaMorphoNotToken(address notToken) public {
+        vm.assume(address(notToken) != address(loanToken));
+        vm.assume(address(notToken) != address(collateralToken));
+
+        vm.expectRevert();
+        new MetaMorpho(OWNER, address(morpho), ConstantsLib.MIN_TIMELOCK, notToken, "MetaMorpho Vault", "MMV");
+    }
+
+    function testDeployMetaMorpho(
```
Comment (Rubilmax) [inline test/forge/DeploymentTest.sol:20]: This test is redundant with `MetaMorphoFactoryTest.testCreateMetaMorpho` (i would simply remove it)

**Review by Rubilmax — CHANGES_REQUESTED** [review]

File: test/forge/DeploymentTest.sol (hunk 1 above)
Comment (MerlinEgalite) [reply test/forge/DeploymentTest.sol:27]: yep

File: test/forge/DeploymentTest.sol (hunk 2 above)
Comment (MerlinEgalite) [reply test/forge/DeploymentTest.sol:20]: One is testing that `MetaMorphoFactory` is working as expected, the other is testing that deploying a MM works as expected. I think this is a little bit different no?

Comment (Rubilmax) [reply test/forge/DeploymentTest.sol:20]: Alright!

**Review by Rubilmax — APPROVED** [review]

**Review by Rubilmax — APPROVED** [review]

**Review by Jean-Grimal — APPROVED** [review]

**Pull request merged**